src/components/Dialogs/Conditions/ConditionsManager.js

```diff
@@ -0,0 +1,165 @@
+        parseConditionsStringToArray(conditions)
+    )
+
+    const addCondition = () => setConditionsList([...conditionsList, ''])
+
+    const removeCondition = id =>
+        setConditionsList(conditionsList.filter((_, index) => index !== id))
+
+    const setCondition = (id, value) =>
+        setConditionsList(
+            conditionsList.map((condition, index) =>
+                index === id ? value : condition
+            )
+        )
+
+    const renderModalTitle = () =>
+        isInLayout
+            ? i18n.t('Edit dimension: {{dimensionName}}', {
+                  dimensionName: dimension.name,
+                  nsSeparator: '^^',
+              })
+            : i18n.t('Add dimension: {{dimensionName}}', {
+                  dimensionName: dimension.name,
+                  nsSeparator: '^^',
+              })
+
+    const renderModalContent = () => (
+        <>
+            <div>
+                <p className={classes.paragraph}>
+                    {i18n.t(
+                        'Show items that meet the following conditions for this data item:'
+                    )}
+                </p>
+            </div>
+            <div className={classes.mainSection}>
+                {!conditionsList.length ? (
+                    <p className={classes.paragraph}>
+                        <span className={classes.infoIcon}>
+                            <IconInfo16 />
+                        </span>
+                        {i18n.t(
+                            'No conditions yet, so all values will be included. Add a condition to filter results.'
+                        )}
+                    </p>
+                ) : (
+                    conditionsList.map((condition, index) => (
+                        <div key={index}>
+                            <NumericCondition
+                                condition={condition}
+                                onChange={value => setCondition(index, value)}
+                                onRemove={() => removeCondition(index)}
+                            />
+                            {conditionsList.length > 1 &&
+                                index < conditionsList.length - 1 && (
+                                    <span className={classes.separator}>
+                                        {i18n.t('and')}
+                                    </span>
+                                )}
+                        </div>
+                    ))
+                )}
+                <Button
+                    type="button"
+                    small
+                    onClick={addCondition}
+                    dataTest={'conditions-manager-add-condition'}
+                    className={classes.addConditionButton}
+                >
+                    {conditionsList.length
+                        ? i18n.t('Add another condition')
+                        : i18n.t('Add a condition')}
+                </Button>
+            </div>
+        </>
+    )
```

Comment:
See my other comment about leveraging the `children` prop in the `DimensionModal` for this.